homeassistant/components/aquacell/__init__.py

```diff
@@ -0,0 +1,41 @@
+"""The Aquacell integration."""
+
+from __future__ import annotations
+
+from aioaquacell import AquacellApi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DATA_AQUACELL, DOMAIN
+from .coordinator import AquacellCoordinator
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Aquacell from a config entry."""
+    session = async_get_clientsession(hass)
+
+    aquacell_api = AquacellApi(session)
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {DATA_AQUACELL: aquacell_api}
+
+    coordinator = AquacellCoordinator(hass, aquacell_api, entry)
+
+    await coordinator.async_config_entry_first_refresh()
+    hass.data[DOMAIN][entry.entry_id] = coordinator
```

Comment:
You can use [config entry runtime](https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry) data instead.